src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php

```diff
@@ -88,6 +90,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
     private function initContext()
     {
         require_once $this->getContainer()->get('kernel')->getRootDir() . '/../config/config.inc.php';
+        /** @var LegacyContext $legacyContext */
+        $legacyContext = $this->getContainer()->get('prestashop.adapter.legacy.context');
+        //We need to have an employee or the module hooks don't work
+        //see LegacyHookSubscriber
+        if (!$legacyContext->getContext()->employee) {
+            $legacyContext->getContext()->employee = new Employee(1);
```

Comment:
And what happened if the employee with id 1 not exists?

Reply:
I think we have bigger issues then ^^
No it actually comes from the Install class https://github.com/PrestaShop/PrestaShop/blob/7b87a098090620f62bb4e67040c50c0810a047e6/src/PrestaShopBundle/Install/Install.php#L416
This employee should always exist as it is created during the install process as the admin employee: https://github.com/PrestaShop/PrestaShop/blob/7b87a098090620f62bb4e67040c50c0810a047e6/src/PrestaShopBundle/Install/Install.php#L831

But I admit it is mainly a workaround to make the LegacyHookSubscriber work: https://github.com/PrestaShop/PrestaShop/blob/7b87a098090620f62bb4e67040c50c0810a047e6/src/Adapter/LegacyHookSubscriber.php#L229

I don't understand the necessity of this test.. the listeners themselves should be resilient to wether the context is initialized or not but it should not block the WHOLE listener system..
I guess it was made this way because `context`, `context->shop` and `context->employee` are too coupled throughout the code

But I'm afraid of changing this check as I'm sure it will generate lots of side effects!

Reply:
It is the installation process, but your users can change :) 
If you create a new superadmin and remove the old one, the user with `Employee(1)` does not exists anymore.
![image](https://user-images.githubusercontent.com/1462701/51257478-fe686c80-19a7-11e9-8f06-09577cd22574.png)

Reply:
true, and you could also remove all Employees an then I think nothing in Prestashop works any more 😅
the thing is I don't even need the employee or the shop for this feature, I'm juste blocked because of how the code is coupled..
What would you suggest? Removing the check on employee in the `LegacyHookSubscriber`? trying to fetch a random employee just to fill the variable?

Reply:
Finding the first super admin employee you can have :) Otherwise in my case I will not be able to run the script because user_id=1 is not found :/

Reply:
well actually if no employee matches the id it doesn't throw any exception or error..
so I used 42! 😀